include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,151 @@
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct callable : decorated_with<OptionsValues, Options...>
+  {
+    using base = decorated_with<OptionsValues, Options...>;
+
+    template<typename T>
+    constexpr auto operator[](T t) const
+    requires( requires(base const& b) { b[t];} )
+    {
+      auto new_traits = base::operator[](t);
+      return  Func<decltype(new_traits)>{new_traits};
+    }
+
+    template<typename T> void operator[](T t) const =delete;
+
+    template<typename... Args>
+    constexpr auto behavior(auto arch, Args&&... args) const
+    {
+      return Func<OptionsValues>::deferred_call(arch, EVE_FWD(args)...);
+    }
+
+    protected:
+    constexpr Func<OptionsValues> const& derived() const { return static_cast<Func<OptionsValues>>(*this); }
+  };
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  {
+    template<callable_options O, typename T, typename... Ts>
+    constexpr auto behavior(auto arch, O const& opts, T x0,  Ts const&... xs) const
+    {
+      using func_t                =  Func<OptionsValues>;
+      constexpr bool is_supported = requires{ func_t::deferred_call(arch, opts, x0, xs...); };
+      constexpr bool any_simd     = (simd_value<T> || ... || simd_value<Ts>);
+
+      // Are we dealing with a no-condition call ?
+      if constexpr( option_type_is<condition_key, O, ignore_none_> )
+      {
+        // If a deferred call is present, let's call it
+        if      constexpr( is_supported )  return func_t::deferred_call(arch, opts, x0, xs...);
+        else if constexpr( any_simd )
+        {
+          // if not, try to slice/aggregate from smaller wides or to map the scalar version
+          if constexpr((has_aggregated_abi_v<Ts> || ...)) return aggregate(this->derived(), x0, xs...);
+          else                                            return map(this->derived(), x0, xs...);
+        }
+      }
+      else
+      {
+        // Grab the condition
+        auto const cond     = opts[condition_key];
+
+        // Drop the conditional and process the function call with the universal masker
+        auto const rmv_cond = options{rbr::drop(condition_key, opts)};
+        auto const f        = Func<decltype(rmv_cond)>{rmv_cond};
+
+        // If the conditional call is supported, call it
+        if      constexpr( is_supported ) return detail::mask_op(cond, f, x0, xs...);
```

Comment:
I am not sure if I understand how you plan to support conditionals.

You say `is_supported` for a conditional call. What would that spec look like?

Alternatively - it should always be supported and this just keeps being in all overloads.

Reply:
Yup this is fishy, removign elementwise for now as discussed